bokehjs/src/coffee/models/ranges/data_range1d.coffee

```diff
@@ -125,9 +136,17 @@ export class DataRange1d extends DataRange
     [start, end] = @_compute_range(min, max)
 
     if @_initial_start?
-      start = @_initial_start
+      if @_mapper_type == "log"
+        if @_initial_start > 0
+          start = @_initial_start
+      else
+        start = @_initial_start
     if @_initial_end?
-      end = @_initial_end
+      if @_mapper_type == "log"
+        if @_initial_end > 0
+          end = @_initial_end
+      else
+        end = @_initial_end
```

Comment:
The effect of these two blocks is to silently ignore any values less than zero in the presence of lag mappers, correct?  A log message is probably warranted